server/serverAPI/controller/InventoryController.ts

```diff
@@ -1,318 +1,184 @@
-            ["_id", req.params.userID]
-        ]);
-
-        let user: ISensitiveUser | null;
-        try {
-            user = await this.database.Get(parameters);
-        } catch (error) {
-            res.status(400)
-                .json(ResponseFormatter.formatAsJSON(ResponseTypes.ERROR, this.getException(error)));
-            return;
-        }
-
-        if (user === null) {
-            res.status(404)
-                .json(ResponseFormatter.formatAsJSON(ResponseTypes.ERROR, "User hasn't been found."));
-            return;
-        }
+    getAll = async (req: Request, res: Response) => {
+        let parameters = new Map<string, any>([["username", req.serverUser.username]]);
 
-        res.status(200)
-            .json(ResponseFormatter.formatAsJSON(ResponseTypes.SUCCESS, user.inventory));
+        return this.requestGet(parameters, res).then(user => {
+            return this.sendSuccess(200, res, user.inventory);
+        }, (response) => response);
     }
 
     /**
-    * Adds food to user's inventory where user is at specified userID.
-    * Upon successful operation, this handler will return all food items from user's inventory.
-    * 
-    * @param req Request parameter that holds information about request.
-    * @param res Response parameter that holds information about response.
-    */
-    addFood = async (req: Request, res: Response) => {
-        let parameters = new Map<string, any>([
-            ["_id", req.params.userID]
-        ]);
-
-        let user: IInternalUser | null;
-        try {
-            user = await this.database.Get(parameters);
-        } catch (error) {
-            res.status(400)
-                .json(ResponseFormatter.formatAsJSON(ResponseTypes.ERROR, this.getException(error)));
-            return;
-        }
-
-        if (user === null) {
-            res.status(404)
-                .json(ResponseFormatter.formatAsJSON(ResponseTypes.ERROR, "User hasn't been found"));
-            return;
-        }
+     * Adds food to user's inventory.
+     * 
+     * @param req Request parameter that holds information about request.
+     * @param res Response parameter that holds information about response.
+     */
+    add = async (req: Request, res: Response) => {
+        let parameters = new Map<string, any>([["username", req.serverUser.username]]);
 
-        let nutrients: string = req.body.nutrients === undefined ? "[]" : req.body.nutrients;
+        return this.requestGet(parameters, res).then(async user => {
+            let nutrients = this.parseNutrients(req.body.nutrients);
 
-        if (nutrients.at(0) !== '[') {
-            nutrients = "[" + nutrients + "]";
-        }
+            let ingredientSchema = new InventoryIngredientSchema(
+                Number.parseInt(req.body?.id),
+                req.body?.name,
+                req.body?.category,
+                nutrients,
+                Number.parseFloat(req.body?.expirationDate)
```

Comment:
I realize maybe I should have made this comment elsewhere, but I'm just noticing that we use a number for the expiration date. Should we instead use a `Date` type?

Reply:
no, and I will explain you why. Essentially, it's a good practice to store date in number format in database. If we store it as number we are essentially having carte blanche on how we interpret data to the front-end... like different formats and stuff. So that's the reason for keeping it as a number. It simply gives more flexibility.

Reply:
Got it